sentry/src/main/java/io/sentry/instrumentation/file/FileIOSpanManager.java

```diff
@@ -0,0 +1,100 @@
+package io.sentry.instrumentation.file;
+
+import io.sentry.ISpan;
+import io.sentry.Sentry;
+import io.sentry.SpanStatus;
+import io.sentry.util.StringUtils;
+import java.io.Closeable;
+import java.io.File;
+import java.io.IOException;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+final class FileIOSpanManager {
+
+  private final @Nullable ISpan currentSpan;
+  private final @Nullable File file;
+
+  private @Nullable Throwable throwable = null;
+  private @NotNull SpanStatus spanStatus = SpanStatus.OK;
+  private long byteCount;
+
+  static @Nullable ISpan startSpan(final @NotNull String op) {
+    final ISpan parent = Sentry.getSpan();
```

Comment:
to make it testable, make a ctor with IHub and without, that defaults to `HubAdapter.getInstance()`, see eg SentryOkHttpInterceptor